devlib/target.py

```diff
@@ -701,6 +703,20 @@ def _install_module(self, mod, **params):
     def _resolve_paths(self):
         raise NotImplementedError()
 
+    def is_network_connected(self):
+        self.logger.debug('Checking for internet connectivity...')
+
+        ip_address = GOOGLE_DNS_SERVER_ADDRESS
+
+        command = 'ping -q -c 1 -w 1 {} 2>&1'.format(ip_address)
+
+        output = self.execute(command, check_exit_code=False)
+        if 'network is unreachable' in output.lower():
+            self.logger.debug('Cannot find IP address {}'.format(ip_address))
+            return False
+        else:
```

Comment:
What if the command fails for some other reason (e.g. ping is not on target)? Should probably check that the command succeed, as you're disabling `check_exit_code`. Also, would probably be a good idea to invoke ping via busybox...

Reply:
>  Should probably check that the command succeed

Hmm, yeah we should probably check we were actually able to execute `ping`. I just don't want to introduce a failure if the ICMP packet happens to get lost or delayed, and I don't want to introduce a delay by sending several. I'll see what I can think of.

>  Also, would probably be a good idea to invoke ping via busybox...

Yeah.. the problem is that we need root to ping - the `ping` command typically has `setuid` set.

Reply:
> Yeah.. the problem is that we need root to ping - the ping command typically has setuid set.

Sorry, I'm not sure I understand this. Could you elaborate? Why do you need root?

Reply:
[You need a certain capability to open the type of socket that you need to send ICMP packets](https://superuser.com/a/1035983/446515).

Reply:
Ah, ok, I see. That makes sense. Thanks.